### PR TITLE
DAOS-13428 test: Fix expected bdev roles for pmem config

### PR DESCRIPTION
**Pull request by phender**
When run with a storage config using pmem, the
control/dmg_storage_query.py test should still expect bdevs to have a 'data' role assigned.

Also resolving TestContainer.list_snaps conflict introduced with the master merge.

Skip-unit-tests: true
Skip-fault-injection-test: true
Allow-unstable-test: true
Test-tag: test_replay_snapshots test_dmg_storage_query_devices

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [x] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/server/replay.py
Comment (daltonbohning) [inline src/tests/ftest/server/replay.py:111]: Is the status check necessary? Since `CommandFailure` is raised if status != 0

**Review by daltonbohning — APPROVED** [review]

**Comment by phender** [comment]
Both modified tests passed in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12148/2/https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12148/2/

File: src/tests/ftest/server/replay.py
Comment (daltonbohning) [inline src/tests/ftest/server/replay.py:110]: FYI, this is how I've been using it
```suggestion
        detected = [entry["epoch"] for entry in self.container.list_snaps()["response"]]
```
But that does beg the question of whether we should have a "simple" return, if that's what tests want

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — DISMISSED** [review]
The `list_snaps` change statically looks safe, but we probably should have ran `DaosSnapshotTest` too for good measure

**Review by tanabarr — APPROVED** [review]

File: src/tests/ftest/control/dmg_storage_query.py
Comment (tanabarr) [reply src/tests/ftest/control/dmg_storage_query.py:53]: I'm hoping we will be able to land PR-11978 (just waiting for one more review from @kjacque or @mjmac then I will request a force landing) shortly so would make sense just to remove the block and land after.

Comment (phender) [reply src/tests/ftest/control/dmg_storage_query.py:53]: Ok, I'll wait for that landing and then update this PR to remove the no-longer-needed code.

**Review by daltonbohning — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]
fyi the go fmt error occurs with go version 1.19 but not with 1.18

**Comment by phender** [comment]
All tests passed https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12148/5/testReport/

**Pull request merged**
